pkg/virtualfs/file_reader.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package virtualfs
+
+import (
+	"io"
+
+	"github.com/kopia/kopia/fs"
+
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/log"
+)
+
+// ReaderSeekerCloser implements io.Reader, io.Seeker and io.Closer
+type ReaderSeekerCloser interface {
+	io.Reader
+	io.Seeker
+	io.Closer
+}
+
+type readerSeekerCloser struct {
```

Comment:
The struct names are a little confusing, they don't convey the intent.
Adding a couple of comments helps. At a later time, somebody else will try to understand what this is intended to do, and they do not have the same context you have in your mind right now.

```suggestion
// readSeekerWrapper adds a dummy Close method to a ReadSeeker
type readSeekerWrapper struct {
```